### PR TITLE
Cosmos API review changes

### PR DESCRIPTION
**Pull request by ajcvickers**
* Made `GetCosmosPartitionKeyStoreName` internal
* `GetCosmosDefaultContainerName` -> `GetCosmosDefaultContainer`
* Remove `Cosmos` from public namespaces to align with core
* Make types internal to align with other providers

Also, enabled API consistency tests and added some doc comments

### CODE REVIEW

**Comment by ajcvickers** [comment]
Team: since Andriy is out, could one of you sign off on this. I don't think there is anything controversial.

**Comment by ajcvickers** [comment]
I'm going to merge this to unblock work on the query API review changes.

@AndriySvyryd Please take a look when you get back.

**Comment by smitpatel** [comment]
Looking into this.

File: src/EFCore.Cosmos/Extensions/CosmosDbContextOptionsExtensions.cs [hunk 1]
```diff
@@ -3,16 +3,28 @@
...
 // ReSharper disable once CheckNamespace
 namespace Microsoft.EntityFrameworkCore
 {
+    /// <summary>
+    ///     Cosmos-specific extension methods for <see cref="DbContextOptionsBuilder" />.
+    /// </summary>
     public static class CosmosDbContextOptionsExtensions
     {
+        /// <summary>
+        ///     Configures the context to connect to a Azure Cosmos database.
```
Comment (smitpatel) [inline src/EFCore.Cosmos/Extensions/CosmosDbContextOptionsExtensions.cs]: a or an?

**Comment by ajcvickers** [comment]
Thanks @smitpatel

**Comment by AndriySvyryd** [comment]
:shipit:

**Pull request merged**